src/screens/ChatScreen.tsx

```diff
@@ -0,0 +1,478 @@
+    }
+    return (
+      <View style={styles.suggestionContainer}>
+        {chatMembers
+          .filter(({ name }) =>
+            name.toLowerCase().includes(keyword.toLowerCase()),
+          )
+          .map(({ id, name }) => (
+            <Pressable
+              key={id}
+              onPress={() => onSuggestionPress({ id, name })}
+              style={{ padding: 10 }}
+            >
+              <Text>{name}</Text>
+            </Pressable>
+          ))}
+      </View>
+    );
+  };
+
+  const renderComposer = (props: any) => {
+    const { text } = props;
+    return (
+      <View style={styles.composerContainer}>
+        {isEditMessage && (
+          <View
+            {...props}
+            style={{
+              width: 60,
+              alignItems: 'center',
+              justifyContent: 'center',
+            }}
+          >
+            <MaterialIcons
+              name="cancel"
+              size={24}
+              color={PRIMARY_COLOR}
+              onPress={() => handleCancelEditMessage()}
+            />
+          </View>
+        )}
+
+        <View style={styles.inputContainer}>
+          <MentionInput
+            {...props}
+            placeholder={t('Type somenthing....')}
+            style={{ paddingLeft: 10 }}
+            value={text}
+            onChange={(value) => setInputMessage(value)}
+            partTypes={[
+              {
+                trigger: MENTION_CHAT_TRIGGER,
+                renderSuggestions,
+                textStyle: { fontWeight: 'bold', color: PRIMARY_COLOR },
+                getPlainString: (mention) => getPlainString(mention),
+              },
+            ]}
+          />
+        </View>
+        <View
+          {...props}
+          style={{
+            width: 60,
+            alignItems: 'center',
+            justifyContent: 'center',
+          }}
+        >
+          <Send
+            {...props}
+            disabled={!text}
+            containerStyle={styles.sendButtonContainer}
+          >
+            <MaterialIcons
+              {...props}
+              name="send"
+              size={24}
+              color={PRIMARY_COLOR}
+            />
+          </Send>
+        </View>
+      </View>
+    );
+  };
+
+  const handleCancelEditMessage = () => {
+    setInputMessage('');
+    setIsEditMessage(false);
+    setMessageSelected(null);
+  };
+
+  const chatMentionsReplacer = (text: string) => {
+    const userId = text.match(MENTION_REGEX)?.groups?.id;
+    const userName =
+      itemMemberships?.find(({ member }) => member.id === userId)?.member
+        .name || UNKNOWN_CHAT_MEMBER;
+    return MENTION_CHAT_TRIGGER + userName;
+  };
+
+  const rules = {
+    code_inline: (node: any) => {
+      const messageTextMentions = node.content.replace(
+        MENTION_REGEX,
+        chatMentionsReplacer,
+      );
+      return <Text key={node.key}>{messageTextMentions}</Text>;
+    },
+  };
+
+  const renderMessageText = (props: any) => {
```

Comment:
Avoid any.
Also this file is way too big. Could you create a `components/chatbox` folder and split this file in smaller components? Each render functions, and probably it would be nice to split the buttons as well -> each button will then contain their own hook.